LayerClient/LayerClient.py

```diff
@@ -0,0 +1,509 @@
+                    'metadata': metadata,
+                }
+            )
+        )
+
+    def send_message(self, conversation, sender, message_parts,
+                     notification=None):
+        """
+        Send a message to a conversation.
+
+        Parameters:
+        - `conversation`: A `LayerClient.Conversation` instance for the
+        conversation we wish to send to
+        - `sender`: A `LayerClient.Sender` instance
+        - `message_parts`: An array of `LayerClient.MessagePart` objects
+        - `notification`: Optional `PushNotification` instance.
+        """
+        request_data = {
+            'sender': sender.as_dict(),
+            'parts': [
+                part.as_dict() for part in message_parts
+            ],
+        }
+        if notification:
+            request_data['notification'] = notification.as_dict()
+
+        return Message.from_dict(
+            self._raw_request(
+                METHOD_POST,
+                self._get_layer_uri(
+                    LAYER_URI_CONVERSATIONS,
+                    conversation.uuid(),
+                    LAYER_URI_MESSAGES,
+                ),
+                request_data,
+            )
+        )
+
+    def send_announcement(self, sender, recipients, message_parts,
+                          notification=None):
+        """
+        Send an announcement to a list of users.
+
+        Parameters:
+        - `sender`: A `LayerClient.Sender` instance. The sender must have a
+            name, as this endpoint cannot be used with a sender ID.
+        - 'recipients`: A list of strings, each of which is a recipient ID.
+        - `message_parts`: An array of `LayerClient.MessagePart` objects
+        - `notification`: Optional `PushNotification` instance.
+        """
+        request_data = {
+            'sender': {
+                'name': sender.name,
+            },
+            'parts': [
+                part.as_dict() for part in message_parts
+            ],
+            'recipients': recipients,
+        }
+        if notification:
+            request_data['notification'] = notification.as_dict()
+
+        return Announcement.from_dict(
+            self._raw_request(
+                METHOD_POST,
+                self._get_layer_uri(
+                    LAYER_URI_ANNOUNCEMENTS,
+                ),
+                request_data,
+            )
+        )
+
+
+class Announcement(BaseLayerResponse):
+    """
+    Contains the data returned from the API when sending an Announcement
+    """
+
+    def __init__(self, id, url, sent_at, recipients, sender, parts):
+        self.id = id
+        self.url = url
+        self.sent_at = sent_at
+        self.recipients = recipients
+        self.sender = sender
+        self.parts = parts
+
+    @staticmethod
+    def from_dict(dict_data):
+        sent_at = (
+            dateutil.parser.parse(dict_data.get('sent_at'))
+            if dict_data.get('sent_at') else None
+        )
```

Comment:
because of the line length restriction this inline-conditional is kind of hard to read. might be better broken into a traditional multi-line if-else